fri/server/main.py

```diff
@@ -47,6 +48,59 @@ def upload(dir):
         resp.status_code = 500
         return resp
 
+@app.route('/build/<dir>', methods=['POST'])
+def build(dir):
+    makestudy_dir = "demo" + "/" + dir
+    cur_path = os.getcwd()
+    concore_path = os.path.abspath(os.path.join(cur_path, '../../'))
+    dir_path = os.path.abspath(os.path.join(concore_path, dir))

```

Comment:
"dir" here must be replaced by a splitted version to get the "sample" below:
split by "/"
as in dir = demo/sample <--- we want the "sample" here.